File: android/src/main/kotlin/com/example/vtb_device_info/helper/DeviceInfoHelper.kt
package com.example.vtb_device_info.helper

import android.annotation.SuppressLint
import android.content.Context
import android.os.Build
import android.provider.Settings.Secure.ANDROID_ID
import android.provider.Settings.Secure.getString

object DeviceInfoHelper {
    @SuppressLint("HardwareIds")
    fun getDeviceInfo(context: Context): Map<String, String> {
        return mapOf(
            "deviceId" to getString(context.contentResolver, ANDROID_ID),
            "deviceName" to Build.MODEL,
            "deviceModel" to Build.PRODUCT,
            "systemName" to "Android",
            "systemVersion" to Build.VERSION.RELEASE
        )
    }
}

File: ios/Classes/Helpers/DeviceInfoHelper.swift
//
//  DeviceInfoHelper.swift
//  Pods
//
//  Created by Duc Pham on 20/11/24.
//

import UIKit

class DeviceInfoHelper {
    func getDeviceInfo() -> [String: String] {
        return [
            "deviceId": UIDevice.current.identifierForVendor?.uuidString ?? "Unknown",
            "deviceName": UIDevice.current.name,
            "deviceModel": UIDevice.current.model,
            "systemName": UIDevice.current.systemName,
            "systemVersion": UIDevice.current.systemVersion,
        ]
    }
}
